docs/UserGuide.md

```diff
@@ -135,13 +138,48 @@ Examples: Lets say your contact list contains the following contacts: **John Doe
 -   `find Alex Lee` would return `Alex Lee`
 -   `find Lee` would return `John Lee` as well as `Alex Lee`
 
+### Undoing a command : `undo`
+
+_This command allows you to undo your last command, restoring ConnectCare application to its previous state._
+
+Format: `undo`
+
+<box type="tip" seamless>
+
+**Tip:** 
+The undo command does not undo every single command, only those that change the address-book in a significant way. 
+For a comprehensive deep-dive into the undo command, please refer to the [Implementation section of our Developer Guide](https://ay2324s2-cs2103t-w12-4.github.io/tp/DeveloperGuide.html) 
+
+</box>
+
+_IMAGE_BEFORE and IMAGE_AFTER TBA_
+
+### Redoing a command : `redo`
+
+_This command allows you to redo your command, restoring ConnectCare application to its previous state._
```

Comment:
Undo and Redo sound like the same command. Maybe it could be slightly rephrased.
Something like "restores the change that was undone" or "reverses your last undo"?